src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -285,19 +288,38 @@ export const selectAggregatedEarnUserStakingOpportunityByStakingId = createDeepE
   selectAggregatedUserStakingOpportunityByStakingId,
   selectMarketDataSortedByMarketCap,
   selectAssets,
-  (opportunity, marketData, assets): StakingEarnOpportunityType | undefined =>
-    opportunity &&
-    Object.assign({}, STAKING_EARN_OPPORTUNITIES[opportunity.assetId], opportunity, {
-      chainId: fromAssetId(opportunity.assetId).chainId,
-      fiatAmount: bnOrZero(opportunity.stakedAmountCryptoBaseUnit)
-        .times(marketData[opportunity.underlyingAssetId as AssetId]?.price ?? '0')
-        .toString(),
-      isLoaded: true,
-      icons: opportunity.underlyingAssetIds
-        .map(assetId => assets[assetId]?.icon)
-        .map(icon => icon ?? ''),
-      opportunityName: opportunity.name,
-    }),
+  (opportunity, marketData, assets): StakingEarnOpportunityType | undefined => {
+    if (!opportunity) return
+
+    const asset = assets[opportunity.assetId]
+    const underlyingAsset = assets[opportunity.underlyingAssetId]
+
+    const aggregatedEarnUserStakingOpportunity: StakingEarnOpportunityType = Object.assign(
```

Comment:
Assigned to a const for type assignment purposes, nothing really changed here except for the addition of `cryptoAmountBaseUnit` and `cryptoAmountPrecision` for types completeness